### DISCUSSION TITLE
Lesson 11: cast send returns deserialization error

### DISCUSSION THREAD

**Discussion by LanceAddison** [Cyfrin/foundry-full-course-cu#1771] [Q&A]
I deployed my MoodNft contract to the Sepolia testnet and when I go to the contract on etherscan and interact with it there functions like mintNft(), safeTransferFrom(), and flipMood() all work but when I try and do those from the command line in vs code I get this deserialization error. 

Here is my command with the error: 
```
cast send 0xC29b1fFCF035d18F94823B08748c9363a8B306d9 "min
tNft()" --rpc-url $SEPOLIA_RPC_URL --private-key $PRIVATE_KEY
Error: 
deserialization error: invalid number at line 1 column 574

Context:
- invalid number at line 1 column 554
```

I checked that all the details are correct and my rpc url is using Alchemy but I get the same error with any other function that requires changing the state of the blockchain like flipMood. I also ran this command on anvil only changing the rpc url and private key and it worked perfectly fine. 

It's not a big deal since the contract technically still works but if anyone knows why this is happening and possibly how I can fix it that would be great.

**Comment by EngrPips** [comment]
Hello @LanceAddison, Did we resolve this already on Discord?

**Reply by LanceAddison** [reply]
I don't think so but I'm not sure. I'm not sure where it would even be on discord.

**Reply by EngrPips** [reply]
Hey @LanceAddison, are you trying to mint a Basic NFT or the Mood NFT?

**Reply by LanceAddison** [reply]
I am trying to mint the Mood Nft. It mints fine when using etherscan but when I try to mint with the cast command above I get the deserialization error. My project is on my profile if it's any help.

**Comment by theirrationalone** [comment]
just update your `foundry framework` and everything will be okay back again.

Yep, You're doing it absolutely right, However the problem is `foundry's cast` version (some bugs might stale up the foundry). 

I know that you already know how to update foundry better than me. However, you can follow up steps given below. 😅

1. Update your foundry: 

```bash
foundryup
```

2. delete the current cli.
3. open new bash cli.
4. enable the env vars...

```bash
source .env
```

5. Now you're good to go. Try to re-interact with your smart contract using that `cast` cmds again.

;`)

**Reply by EngrPips** [reply]
Excellent, thanks for this heads up, buddy.
